### DISCUSSION TITLE
File content privacy

### DISCUSSION THREAD

**Discussion by nickpascucci** [zed-industries/zed#6492] [Feature Requests]
In [Zed's FAQ](https://zed.dev/faq#:~:text=their%20status%20messages.-,file%20contents,-%2D%20When%20you%27re%20sharing) it's stated that file contents are proxied to peers through Zed's servers:

> **file contents** - When you're sharing a project in a call, your collaborators can open any file in your project, as well as files that are returned by your language servers from requests like 'go to definition' and 'find all references'. When they do open these files, we send them to our servers, which forward them to the collaborator.

Can you elaborate on what protections are in place for that data? Are file contents sent in the clear, or is there some form of end-to-end encryption? What are the retention policies for file contents on Zed's servers?

**Comment by tl-carlos-rosa** [comment]
Would also like to know this, the feature looks cool but we need more information before being able to use it in a company environment.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
